File: day4/ans.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        insert_idx = 0
        for idx, num in enumerate(nums):
            if num != 0:
                if idx > insert_idx:
                    nums[insert_idx], nums[idx] = nums[idx], nums[insert_idx]
                
                insert_idx += 1

File: day4/ans.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        
        int insert_idx = 0;
        for(unsigned int current_idx = 0; current_idx < nums.size(); current_idx++) {
            if(nums[current_idx] != 0) {
                if(current_idx > insert_idx) {
                    swap(nums[insert_idx], nums[current_idx]);
                }
                insert_idx += 1;
            }
        }
    }
};
